src/libsyntax/codemap.rs

```diff
@@ -126,12 +125,12 @@ impl StableFilemapId {
 //
 
 pub struct CodeMap {
-    pub(super) files: RefCell<Vec<Lrc<FileMap>>>,
-    file_loader: Box<FileLoader>,
+    pub(super) files: Lock<Vec<Lrc<FileMap>>>,
+    file_loader: Box<FileLoader + Sync + Send>,
     // This is used to apply the file path remapping as specified via
     // --remap-path-prefix to all FileMaps allocated within this CodeMap.
     path_mapping: FilePathMapping,
-    stable_id_to_filemap: RefCell<FxHashMap<StableFilemapId, Lrc<FileMap>>>,
+    stable_id_to_filemap: Lock<FxHashMap<StableFilemapId, Lrc<FileMap>>>,
```

Comment:
I think it would make sense to put `stable_id_to_filemap` into the same `Lock` as `files` since they should be modified atomically.

Reply:
We could probably also remove `files` altogether since `stable_id_to_filemap` seems to contain the same information?

Reply:
We address `FileMaps` by their index in the `files` array quite a bit. For now, I would just keep both data structures.